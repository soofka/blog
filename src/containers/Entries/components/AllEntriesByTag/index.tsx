import * as React from 'react';
import { inject } from 'mobx-react';

import RoutingProvider from 'common/RoutingProvider';

import EntriesByTagMeta from 'components/Meta/components/EntriesByTagMeta';

import { Entries, EntriesParamsInterface, EntriesPropsInterface } from 'containers/Entries';
import { EntryInterface } from 'containers/Entry';

export class AllEntriesByTag extends Entries {
  getMeta() {
    const { match: { params: { tag } } } = this.props;
    return <EntriesByTagMeta tag={tag} />;
  }

  filterEntries(entries: EntryInterface[], params: EntriesParamsInterface): EntryInterface[] {
    const { tag } = params;

    return entries.filter((entry: EntryInterface) => {
      return entry.tags.map((entryTag: string) =>
        RoutingProvider.parseTextToNiceUrl(entryTag)).indexOf(prepareTag(tag)) >= 0;
    });
  }
}

function prepareTag(tag: string): string {
  return tag.trim().toLowerCase();
}

export default inject('languageStore')(AllEntriesByTag as any);
