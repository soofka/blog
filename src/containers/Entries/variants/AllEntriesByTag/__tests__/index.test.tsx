import * as React from 'react';
import { shallow } from 'enzyme';

import AllEntriesByTag from 'containers/Entries/variants/AllEntriesByTag';

describe('<AllEntriesByTag/>', () => {

  it('renders correctly', () => {
    expect(shallow(<AllEntriesByTag/>)).toMatchSnapshot();
  });

});
