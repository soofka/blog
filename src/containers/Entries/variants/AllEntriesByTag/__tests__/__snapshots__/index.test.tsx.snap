exports[`<AllEntriesByTag/> renders correctly 1`] = `
<AllEntriesByTag
  languageStore={
    Object {
      "getLanguage": [Function],
      "setLanguage": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "date": "testDate",
        "niceUrl": "testNiceUrl",
        "tag": "testTag",
      },
    }
  } />
`;
